src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -55,27 +55,185 @@ public class RD2_DataMigrationMapper {
      */
     public npe03__Recurring_Donation__c convertToEnhancedRD() {
+        if (isConverted()) {
+            return rd;
+        }
+        
+        setInstallmentFields(rd);
+        setStartDate(rd);
+        setDayOfMonth(rd);
+        setStatus(rd);
+        setOtherFields(rd);
+
         return rd;
     }
 
+    /**	
+     * @description Returns true if the RD record is already in enhanced Recurring Donation format	
+     */	
+    public Boolean isConverted() {	
+        return String.isNotBlank(rd.Status__c) 
+            && rd.InstallmentFrequency__c != null;	
+            //frequency is null for "1st and 15th" installment period even when converted
+            //check if the RD schedule exists //TODO
+    }
+
     /**
-     * @description Generate the RDSchedule Record(s) for the Recurring Donation
-     * @return List<RecurringDonationSchedule__c> Schedule records to insert
+     * @description Sets Installment Frequency and Installment Period fields where applicable
+     * @param rd Recurring Donation in the legacy format
+     * @return void
      */
-    public List<RecurringDonationSchedule__c> getScheduleRecords() {
-        return new List<RecurringDonationSchedule__c>();
+    private void setInstallmentFields(npe03__Recurring_Donation__c rd) {
+
+        Set<String> frequencyOneInstallmentPeriods = new Set<String>{
+            System.Label.npe03.RecurringDonationInstallmentPeriodYearly,
+            System.Label.npe03.RecurringDonationInstallmentPeriodMonthly,
+            System.Label.npe03.RecurringDonationInstallmentPeriodWeekly
+        };
+
+        if (rd.npe03__Installment_Period__c == System.Label.npe03.RecurringDonationInstallmentPeriodQuarterly) {
+            rd.npe03__Installment_Period__c = RD2_Constants.INSTALLMENT_PERIOD_MONTHLY;
+            rd.InstallmentFrequency__c = 3;
+
+        } else if (frequencyOneInstallmentPeriods.contains(rd.npe03__Installment_Period__c)) {
+            if (rd.InstallmentFrequency__c == null) {
+                rd.InstallmentFrequency__c = 1;
+            }
+
+        } else if (rd.npe03__Installment_Period__c != System.Label.npe03.RecurringDonationInstallmentPeriod1stand15th) {
+            handleCustomInstallmentPeriod(rd);
+        }
+
+        if (rd.npe03__Schedule_Type__c == System.Label.npe03.RecurringDonationDivideValue
+            && rd.npe03__Installment_Amount__c != null 
+            && rd.npe03__Installment_Amount__c != 0
+        ) {
+            rd.npe03__Amount__c = rd.npe03__Installment_Amount__c;
+        }
+    }
+
+    /**
+     * @description Sets Installment Frequency and Installment Period for custom Installment Period
+     * @param rd Recurring Donation in the legacy format
+     * @return void
+     */
+    private void handleCustomInstallmentPeriod(npe03__Recurring_Donation__c rd) {
+        npe03__Custom_Installment_Settings__c settings = UTIL_ListCustomSettingsFacade.getMapCustomInstallmentSettings()
```

Comment:
@RadSFDeveloper This looks correct. You're trying to match the installment period period picklist value to a custom installment period and if there is a match then you set the InstallmentPeriod field correctly and apply the installment frequency from the custom installment record. 

When there is no match then an exception is thrown (which will eventually be a log action, but exception can work for now).

I'm not necessarily sure there is anything to fix here.

Reply:
Thanks @force2b for the verification, I have updated tests to set the legacy RD installment period to the custom installment *name* and verified tests pass.